nilearn/regions/tests/test_hierarchical_kmeans_clustering.py

```diff
@@ -112,9 +112,13 @@ def test_hierarchical_k_means():
 
 
 def test_hierarchical_k_means_clustering():
-    data_img, mask_img = generate_fake_fmri(shape=(10, 11, 12), length=15)
+    n_samples = 15
+    n_features = 150
+    data_img, mask_img = generate_fake_fmri(
+        shape=(10, 11, 12), length=n_samples
+    )
     masker = NiftiMasker(mask_img=mask_img).fit()
-    X = masker.transform(data_img).T
```

Comment:
It is indeed worrying, but AFAICT it won't break because in line 135, we only test the first dimension to be equal to n_clusters, which it would always be:
```python
assert X_red.shape[0] == n_clusters
```
The problem would occur when we check the second dimension of `X_red`, and it won't be equal to n_samples we expected